src/js/themes/base.d.ts

```diff
@@ -0,0 +1,825 @@
+        };
+        hover: {
+            background: {
+                dark: {
+                    color: string;
+                    opacity: string;
+                };
+                light: {
+                    color: string;
+                    opacity: string;
+                };
+            };
+            color: {
+                dark: string;
+                light: string;
+            };
+        };
+        input: {
+            padding: string;
+            weight: number;
+        };
+        opacity: {
+            strong: number;
+            medium: number;
+            weak: number;
+        };
+        selected: {
+            background: string;
+            color: string;
+        };
+        spacing: string;
+        size: {
+            xxsmall: string;
+            xsmall: string;
+            small: string;
+            medium: string;
+            large: string;
+            xlarge: string;
+            xxlarge: string;
+            full: string;
+        };
+    };
+    accordion: {
+        icons: {
+            collapse: any;
+            expand: any;
+        };
+    };
+    anchor: {
+        textDecoration: string;
+        fontWeight: number;
+        color: {
+            dark: string;
+            light: string;
+        };
+        hover: {
+            textDecoration: string;
+        };
+    };
+    box: {
+        responsiveBreakpoint: string;
+    };
+    button: {
+        border: {
+            width: string;
+            radius: string;
+        };
+        primary: {};
+        disabled: {
+            opacity: number;
+        };
+        minWidth: string;
+        maxWidth: string;
+        padding: {
+            vertical: string;
+            horizontal: string;
+        };
+    };
+    calendar: {
+        small: {
+            fontSize: string;
+            lineHeight: number;
+            daySize: string;
+            slideDuration: string;
+        };
+        medium: {
+            fontSize: string;
+            lineHeight: number;
+            daySize: string;
+            slideDuration: string;
+        };
+        large: {
+            fontSize: string;
+            lineHeight: number;
+            daySize: string;
+            slideDuration: string;
+        };
+        icons: {
+            previous: any;
+            next: any;
+            small: {
+                previous: any;
+                next: any;
+            };
+        };
+    };
+    carousel: {
+        icons: {
+            current: any;
+            next: any;
+            previous: any;
+        };
+    };
+    checkBox: {
+        border: {
+            color: {
+                dark: string;
+                light: string;
+            };
+            width: string;
+        };
+        check: {
+            extend: string;
+            radius: string;
+            thickness: string;
+        };
+        extend: string;
+        color: string | { dark: string; light: string; };
```

Comment:
can we refactor this to use ColorType https://github.com/grommet/grommet/blob/master/src/js/utils/index.d.ts#L43